### PR TITLE
[Storage Blobs Service] Seal Append Blob

### PR DESCRIPTION
**Pull request by JinmingHu-MSFT**

### CODE REVIEW

File: sdk/storage/inc/blobs/protocol/blob_rest_client.hpp [hunk 1]
```diff
@@ -8176,6 +8222,73 @@ namespace Azure { namespace Storage { namespace Blobs {
...
+        Azure::Core::Nullable<int64_t> AppendPosition;
+      }; // struct SealAppendBlobOptions
+
+      static Azure::Core::Response<SealAppendBlobResult> Seal(
+          const Azure::Core::Context& context,
+          Azure::Core::Http::HttpPipeline& pipeline,
+          const std::string& url,
+          const SealAppendBlobOptions& options)
+      {
+        unused(options);
```
Comment (katmsft) [inline sdk/storage/inc/blobs/protocol/blob_rest_client.hpp:8242]: It's time to delete this right?

File: sdk/storage/test/blobs/blob_container_client_test.cpp [hunk 2]
```diff
@@ -117,6 +117,13 @@ namespace Azure { namespace Storage { namespace Test {
       p1Blobs.insert(blobName);
       p1p2Blobs.insert(blobName);
     }
+    {
+      auto appendBlobClient
+          = m_blobContainerClient->GetAppendBlobClient(RandomString() + "-appendblob");
+      appendBlobClient.Create();
+      auto pageBlobClietn = m_blobContainerClient->GetPageBlobClient(RandomString() + "-pageblob");
+      pageBlobClietn.Create(4096);
```
Comment (katmsft) [inline sdk/storage/test/blobs/blob_container_client_test.cpp:125]: nit: check the spelling.

File: sdk/storage/inc/blobs/protocol/blob_rest_client.hpp (hunk 1 above)
Comment (JinmingHu-MSFT) [reply sdk/storage/inc/blobs/protocol/blob_rest_client.hpp:8242]: I cannot remember which one, but some API doesn't use `options` at all

File: sdk/storage/test/blobs/blob_container_client_test.cpp (hunk 2 above)
Comment (JinmingHu-MSFT) [reply sdk/storage/test/blobs/blob_container_client_test.cpp]: fixed

**Pull request merged**